helm/hpcc/templates/_helpers.tpl

```diff
@@ -45,19 +45,47 @@ Translate a port list to a comma-separated list
  {{- end -}}
 {{- end -}}
 
+{{- define "hpcc.getFirstPlaneForLabel" -}}
+{{- $root := .root -}}
+{{- $label := .label -}}
+{{- $storage := ($root.Values.storage | default dict) -}}
+{{- $planes := ($storage.planes | default list) -}}
+{{- $firstPlane := dict -}}
+{{- range $plane := $planes -}}
+{{- $labels := $plane.labels | default (list "data") -}}
+{{- if (has $label $labels) -}}
+{{- $_ := set $firstPlane "plane" $plane.name -}}
```

Comment:
may be missing something, shouldn't something be checking if set from previous iteration, otherwise $firstPlane.plane is going to be set again and become the last matching plane, not first.

Reply:
yes you're correct, I will fix.